File: api/src/main/kotlin/edu/wgu/osmt/api/model/ApiBatchResult.kt
package edu.wgu.osmt.api.model

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_EMPTY)
data class ApiBatchResult(
    @JsonProperty("success")
    val success: Boolean = false,

    @JsonProperty("message")
    val message: String? = null,

    @JsonProperty("modifiedCount")
    val modifiedCount: Number? = null,

    @JsonProperty("totalCount")
    val totalCount: Number? = null
)

File: ui/src/app/richskill/ApiBatchResult.ts



export interface IBatchResult {
  success?: boolean;
  message?: string;
  modifiedCount?: number;
  totalCount?: number;
}


export class ApiBatchResult implements IBatchResult {
  success?: boolean;
  message?: string;
  modifiedCount?: number;
  totalCount?: number;
  constructor({success, message, modifiedCount, totalCount}: IBatchResult) {
    this.success = success
    this.message = message
    this.modifiedCount = modifiedCount
    this.totalCount = totalCount
  }
}


